src/libs/OptionsListUtils.js

```diff
@@ -636,7 +637,57 @@ function hasEnabledOptions(options) {
 }
 
 /**
- * Build the options for the category tree hierarchy via indents
+ * Sorts categories using a simple object.
+ * It builds an hierarchy (based on an object), where each category has a name and other keys as subcategories.
+ * Via the hierarchy we avoid duplicating and sort categories one by one. Subcategories are being sorted alphabetically.
+ *
+ * @param {Object<String, {name: String, enabled: Boolean}>} categories
+ * @returns {Object[]}
+ */
+function sortCategories(categories) {
+    const hierarchy = {};
+
+    _.each(categories, (category) => {
+        const path = category.name.replaceAll(CONST.PARENT_CHILD_SEPARATOR, '.');
```

Comment:
Is dot (`.`) a forbidden character in the category name?

Reply:
Perhaps, but `path` is not a category name anymore. It is keys for hierarchy's object.

Reply:
What I meant is if it's guranateed that `category.name` doesn't contain dots.

Example: `Movies: Mr. Nobody` -> `Movies` `.` `Mr` `.` `Nobody`

Reply:
If you're not sure that `Mr. Nobody` is not an invalid category name part, it would be a good idea to add such case to unit tests

Reply:
I got you. Actually, a good catch!
We need to test this case 🧐

Reply:
@cubuspl42, I've added logic to handle the case - 21a25fa.

<img width="1728" alt="Screenshot 2023-10-18 at 15 49 28" src="https://github.com/Expensify/App/assets/57314631/21ffff24-76b9-4eca-8f7a-a991b005c856">